internal/client.go

```diff
@@ -1003,6 +1031,54 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	if options.RunID != "" {
+		return nil, errors.New("RunID is not allowed to be used on a prepared update operation")
+	}
+	if options.FirstExecutionRunID != "" {
+		return nil, errors.New("FirstExecutionRunID is not allowed to be used on a prepared update operation")
+	}
+	return &UpdateWorkflowOperation{input: input, doneCh: make(chan struct{})}, nil
+}
+
+// Get blocks until a server response has been received; or the context deadline is exceeded.
+func (op *UpdateWorkflowOperation) Get(ctx context.Context) (WorkflowUpdateHandle, error) {
+	if op.done.Load() {
+		return op.handle, op.err
+	}
```

Comment:
```suggestion
```

Not needed IMO (neither is the atomic bool)

Reply:
Ah, right; when the channel is closed, it will unblock immediately. I removed `done` now.

Reply:
Hm, as I was removing `done`, I also extended the integration test exercising the concurrency and it's failing now.